src/protagonist/API.Tests/Integration/CustomHeaderTests.cs

```diff
@@ -0,0 +1,200 @@
+        const int customerId = 90;
+        var path = $"customers/{customerId}/customHeaders";
+        var customHeader = new CustomHeader()
+        {
+            Id = Guid.NewGuid().ToString(),
+            Customer = customerId,
+            Key = "test-key",
+            Value = "test-value"
+        };
+        
+        await dlcsContext.CustomHeaders.AddAsync(customHeader);
+        await dlcsContext.SaveChangesAsync();
+        
+        // Act
+        var response = await httpClient.AsCustomer(customerId).GetAsync(Path.Combine(path, customHeader.Id));
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+    }
+    
+    [Fact]
+    public async Task Get_CustomHeader_404_IfNotFound()
+    {
+        // Arrange
+        const int customerId = 91;
+        var path = $"customers/{customerId}/customHeaders/{Guid.Empty}";
+        
+        // Act
+        var response = await httpClient.AsCustomer(customerId).GetAsync(path);
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+    }
+    
+    [Fact]
+    public async Task Delete_CustomHeader_204()
+    {
+        // Arrange
+        const int customerId = 92;
+        var path = $"customers/{customerId}/customHeaders";
+        var customHeader = new CustomHeader()
+        {
+            Id = Guid.NewGuid().ToString(),
+            Customer = customerId,
+            Key = "test-key",
+            Value = "test-value"
+        };
+        
+        await dlcsContext.CustomHeaders.AddAsync(customHeader);
+        await dlcsContext.SaveChangesAsync();
+        
+        // Act
+        var response = await httpClient.AsCustomer(customerId).DeleteAsync(Path.Combine(path, customHeader.Id));
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.NoContent);
+        var deletedCustomHeader = await dlcsContext.CustomHeaders.AnyAsync(ch => ch.Id == customHeader.Id);
+        deletedCustomHeader.Should().BeFalse();
+    }
+    
+    [Fact]
+    public async Task Get_CustomHeaders_200()
+    {
+        // Arrange
+        const int customerId = 93;
+        var path = $"customers/{customerId}/customHeaders";
+        
+        await dlcsContext.CustomHeaders.AddTestCustomHeader("test-key-1", "test-value-1", customerId);
+        await dlcsContext.CustomHeaders.AddTestCustomHeader("test-key-2", "test-value-2", customerId, space:1);
+        await dlcsContext.CustomHeaders.AddTestCustomHeader("test-key-3", "test-value-3", 1);
+        await dlcsContext.SaveChangesAsync();
+        
+        // Act
+        var response = await httpClient.AsCustomer(customerId).GetAsync(path);
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        
+        var model = await response.ReadAsHydraResponseAsync<HydraCollection<CustomHeader>>();
+        model.Members.Should().HaveCount(2);
+    }
+    
+    [Fact]
+    public async Task Post_CustomHeader_201()
```

Comment:
Do you need tests around the hydra validations? I.e. testing failure - same with what happens when you try and add a duplicate header?  Additionally, what happens when you try and post to a space which doesn't exist?

Reply:
I mainly mean within the controller itself and while the tests themselves are probably not that useful, if you were to output a code coverage report, it would show
```csharp
if (!validationResult.IsValid)
{
    return this.ValidationFailed(validationResult);
}
```
is not covered

In terms of keys, is it worth writing a test to show you can push multiple keys in with the same id? Mainly because if I see key and value, I'd assume it's a dictionary which would mean the keys are unique - having a test like this just makes it cleared that the logic is intentional

And if you're allowing multiple of the same keys, is there a verified order that a key is returned in `GetCustomHeader`? I.e.: is it FIFO or LIFO? and is it worth writing a test to show this?

Reply:
Tests for validation have been added in ddc6b699f722ac3d85cc181ef0fa1930f19baf2b

Reply:
> And if you're allowing multiple of the same keys, is there a verified order that a key is returned in GetCustomHeader? I.e.: is it FIFO or LIFO? and is it worth writing a test to show this?

There is an order of precedence in how the keys are applied in orchestrator but nothing specified about how they should be returned here. Leaving it to DB order is fine for now.

Reply:
a2bba53e489d958f5fa4d980f16cf2ad850df51b adds a test for multiple custom headers using the same ID